### ISSUE TITLE
Files downloaded from the UI should use content-type provided during upload

### ISSUE THREAD

**Issue by mindscratch** [minio/minio#4929]
## Expected Behavior
If an object is uploaded and a Content-Type is provided, when downloading it from the UI the Content-Type should be the same.

## Current Behavior
Currently, when downloading an object from the UI the Content-Type in the response is based off the file extension rather than using the Content-Type provided when uploading the object.

## Possible Solution
The UI should check to see if the object has a Content-Type associated with it and use that in the response, otherwise fall back to using one based on the file extension.

## Your Environment
* Version used (`minio version`): 2017-05-05T01:14:51Z
* Environment name and version (e.g. nginx 1.9.1): docker image

**Comment by krishnasrinivas** [comment]
Thanks for reporting the issue @mindscratch we'll fix it.

**Comment by deekoder** [comment]
We will do this in our next release @mindscratch

**Comment by krishnasrinivas** [comment]
> Currently, when downloading an object from the UI the Content-Type in the response is based off the file extension rather than using the Content-Type provided when uploading the object.

@mindscratch currently the server is not responding with any content-type. Why do you think it responds with content-type based on the object name?

**Comment by mindscratch** [comment]
@krishnasrinivas Previously if I uploaded a file with Content-Type "text/plain" but the file name as "mydoc.pdf", when I would download it from the UI the response header would have Content-Type as "application/pdf", instead of "text/plain". I just started up the latest minio and its behaving differently, when clicking on a file the UI it now makes a request to `http://localhost:9000/minio/webrpc` which returns JSON and then the browser downloads the file. I guess this can be closed.
